{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "618c6015-b4af-46a4-898e-ebf7659ea5a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(GenomicRanges)\n",
    "library(data.table)\n",
    "path=\"/sharedFolder/Data/1_HTGTS/13_Revision1/\"\n",
    "results=\"/sharedFolder/Results/Hotspot/Revision1/Idelalisib/FULL\"\n",
    "previousResults=\"/sharedFolder/Results/Hotspot/Revision1/FULL/6/\"\n",
    "\n",
    "\n",
    "\n",
    "gtf=\"/sharedFolder/Data/gene_coordinatesHG38.tsv\"\n",
    "Database=\"/sharedFolder/Results/Hotspot/Revision1/FULL/6/Database.csv\"\n",
    "gtfT=read.table(gtf,header=TRUE,sep=\"\\t\")\n",
    "gtfG=GRanges(seqnames=as.matrix(gtfT[,1]),ranges=IRanges(start = as.numeric(gtfT[,2])-100,end=as.numeric(gtfT[,3])+100))\n",
    "DatanabaseT=read.table(Database,header=TRUE,sep=\",\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c655621b-29b5-4149-8961-6970be0fef9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#FUNCTIONs\n",
    "modify_vector <- function(vector) {\n",
    "  modified_vector <- vector\n",
    "  for (i in seq_along(vector)) {\n",
    "    if (nchar(vector[i]) >= 10) {\n",
    "      elements <- strsplit(vector[i], \"_\")[[1]]\n",
    "      shortest_index <- which.min(nchar(elements))\n",
    "      modified_vector[i] <- elements[shortest_index]\n",
    "    }\n",
    "  }\n",
    "  return(modified_vector)\n",
    "}\n",
    "\n",
    "circoPlot=function(tempPath,increasedOnly,control,treat){\n",
    "temp=read.table(tempPath,header=TRUE,row.names=1,sep=\",\")\n",
    "\n",
    "if(increasedOnly){\n",
    "temp=temp[which(temp[treat]-temp[control]>0),]\n",
    "}    \n",
    "data=read.table(paste(\"/sharedFolder/Results/Hotspot/Revision1/FULL/6/Database_annotated.csv\",sep=\"\"),header=TRUE,sep=\",\")\n",
    "dataList=c()\n",
    "for(i in rownames(temp)){\n",
    "if(!is.na(grep(i,data[,8])[1])){\n",
    "    dataList=append(dataList,grep(i,data[,8])[1])\n",
    "    }else{\n",
    "    yo=strsplit(i,\"_\")[[1]]\n",
    "    dataList = dataList=append(dataList,which(\n",
    "  data$Chromosome == yo[1] &\n",
    "  data$ChrStart == as.integer(yo[2]) &\n",
    "  data$ChrEnd == as.integer(yo[3])\n",
    ")[1])\n",
    "    }\n",
    "}\n",
    "    \n",
    "data2=data[dataList,]\n",
    "tt=c(\"chr8\",\"126735434\",\"128742951\",\"MYC\")\n",
    "names(tt)=names(data2[,c(1,2,3,8)])\n",
    "label=rbind(data2[,c(1,2,3,8)],t(as.matrix(tt)))\n",
    "condition=c(\"DMSO\",\"Taze\",\"Vale\")\n",
    "forever=matrix(ncol=7)\n",
    "colnames(forever)=c(\"chr1\",\"start1\",\"end1\",\"chr2\",\"start2\",\"end2\",\"score\")\n",
    "label[,1]=gsub(\"chr\",\"hs\",label[,1])\n",
    "nam=c(rownames(temp),\"MYC\")\n",
    "nam[grep(\"chr\",nam)]=\"NA\"\n",
    "label[,4]=modify_vector(nam)\n",
    "\n",
    "write.table(label,paste(results,\"/_label.txt\",sep=\"\"),col.names=FALSE,row.names=FALSE,sep=\"\\t\",quote=FALSE)\n",
    "test=matrix(ncol=7)\n",
    "colnames(test)=c(\"chr1\",\"start1\",\"end1\",\"chr2\",\"start2\",\"end2\",\"score\")\n",
    "extra_info <- cbind(rep(\"chr8\", nrow(data2)), rep(\"126735434\", nrow(data2)), rep(\"128742951\", nrow(data2)))\n",
    "tt=cbind(extra_info,data2[,c(1,2,3)],paste(\"score=\",(temp[,treat]+1)/(temp[,control]+1),sep=\"\"))\n",
    "    colnames(tt)=c(\"chr1\",\"start1\",\"end1\",\"chr2\",\"start2\",\"end2\",\"score\")\n",
    "    forever=rbind(forever,tt)\n",
    "    test=rbind(forever,tt)\n",
    "    test=test[-1,]\n",
    "    \n",
    "    test[,1]=gsub(\"chr\",\"hs\",test[,1])\n",
    "    test[,4]=gsub(\"chr\",\"hs\",test[,4])\n",
    "    write.table(test,paste(results,\"/_link.txt\",sep=\"\"),col.names=FALSE,row.names=FALSE,sep=\"\\t\",quote=FALSE)\n",
    "\n",
    "\n",
    "forever=forever[-1,]\n",
    "\n",
    "# Sostituisci \"chr\" con \"hs\" nelle colonne \"chr1\" e \"chr2\"\n",
    "\n",
    "forever[,1]=gsub(\"chr\",\"hs\",forever[,1])\n",
    "forever[,4]=gsub(\"chr\",\"hs\",forever[,4])\n",
    "\n",
    "write.table(forever,paste(results,\"/_link.txt\",sep=\"\"),col.names=FALSE,row.names=FALSE,sep=\"\\t\",quote=FALSE)\n",
    "setwd(results)\n",
    "system(\"/home/circos-0.69-9/bin/circos -conf /sharedFolder/Circos_conf/yo.conf\")\n",
    "    print((paste(\"mv /sharedFolder/Results/Ligase4/Ligase4.png \",tools::file_path_sans_ext(tempPath),\".png\",sep=\"\")))\n",
    "system(paste(\"mv /sharedFolder/Results/Ligase4/Ligase4.png \",tools::file_path_sans_ext(tempPath),\".png\",sep=\"\")) \n",
    "system(paste(\"mv /sharedFolder/Results/Ligase4/Ligase4.svg \",tools::file_path_sans_ext(tempPath),\".svg\",sep=\"\"))    \n",
    "\n",
    "    system(\"/home/circos-0.69-9/bin/circos -conf /sharedFolder/Circos_conf/yo_NOTHICK.conf\")\n",
    "system(paste(\"mv /sharedFolder/Results/Ligase4/Ligase4.png \",tools::file_path_sans_ext(tempPath),\"_noThick.png\",sep=\"\")) \n",
    "system(paste(\"mv /sharedFolder/Results/Ligase4/Ligase4.svg \",tools::file_path_sans_ext(tempPath),\"_noThick.svg\",sep=\"\"))   \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bf2402d3-c1ff-49a0-86e3-19a0f8660664",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"mv /sharedFolder/Results/Ligase4/Ligase4.png /sharedFolder/Results/Hotspot/Revision1/Idelalisib/FULL/RPMCOmparison/Valerpm_FAIR_ANNOTATED.png\"\n"
     ]
    }
   ],
   "source": [
    "#circoPlot(\"/sharedFolder/Results/Hotspot/Revision1/Idelalisib/FULL/RPMCOmparison/DMSOrpm_FAIR_ANNOTATED6.csv\",FALSE,1,3)\n",
    "#circoPlot(\"/sharedFolder/Results/Hotspot/Revision1/Idelalisib/FULL/RPMCOmparison/Tazerpm_FAIR_ANNOTATED6.csv\",FALSE,1,3)\n",
    "circoPlot(\"/sharedFolder/Results/Hotspot/Revision1/Idelalisib/FULL/RPMCOmparison/Valerpm_FAIR_ANNOTATED.csv\",FALSE,1,3)\n",
    "\n",
    "#circoPlot(\"/sharedFolder/Results/Idelalisib/RPMCOmparison/Tazerpm_FAIR_ANNOTATED.csv\",FALSE,1,3)\n",
    "#circoPlot(\"/sharedFolder/Results/Idelalisib/RPMCOmparison/Valerpm_FAIR_ANNOTATED.csv\",FALSE,1,3)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10f323de-9c5e-4f90-a904-50b4df905756",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bff06a69-4f3c-4ff3-99d2-da3753d41bd3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "bd3e0ad8-f0de-428d-8cc8-ab5e6ccdba54",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf1c591e-a7a8-4f76-a823-b71761e79b56",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
