{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "67eec64f-0147-4ac5-9b3f-57fe6e64ff95",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n"
     ]
    }
   ],
   "source": [
    "noExt=function(x){\n",
    "return(tools::file_path_sans_ext(x))\n",
    "}\n",
    "bnoExt=function(x){\n",
    "return(basename(tools::file_path_sans_ext(x)))\n",
    "}\n",
    "fNumber=function(x){\n",
    "sapply(x,FUN=function(x){\n",
    "strsplit(x,\"__\")[[1]][2]\n",
    "})\n",
    "}\n",
    "rRider=function(x){\n",
    "sapply(x,FUN=function(x){\n",
    "strsplit(x,\"-RIDER\")[[1]][1]\n",
    "})\n",
    "}\n",
    "# centering with 'scale()'\n",
    "center_scale <- function(x) {\n",
    "    scale(x, scale = FALSE)\n",
    "}\n",
    "#LOAD LIBRARIES \n",
    "library(gtools);\n",
    "library(\"GenomicRanges\");\n",
    "options(warn=-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d3102e57-f8cb-4ebd-b84f-c6f7f8768ad2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SETTING PATH \n",
    "pathGroseq=\"/sharedFolder/Data/4_cutTagMet/\"\n",
    "results=\"/sharedFolder/Results/Sequencing\"\n",
    "previousResults=\"/sharedFolder/Results/Hotspot/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "980771de-f4f5-43c0-8f0e-22a139cb570d",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir.create(results,recursive=TRUE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8a644c2f-ef35-4b77-add5-6ed5c9bef806",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtf=read.table(paste(previousResults,\"Database.csv\",sep=\"/\"),header=TRUE,sep=\",\")\n",
    "gtf=gtf[grep(\"Ligase\",gtf[,6]),]\n",
    "a=list.files(pathGroseq,pattern=\".bedgraph\",recursive=TRUE)\n",
    "uu=grep(\"sorted\",a)\n",
    "if(length(uu)!=0){\n",
    "a=a[-uu]\n",
    "    }\n",
    "a=mixedsort(a)\n",
    "a2=sapply(a,FUN=function(x){strsplit(x,\".bedgraph\")[[1]][1]})\n",
    "groseq_count=matrix(0,ncol=length(unique(gtf[,7])),nrow=length(a2))\n",
    "colnames(groseq_count)=sapply(unique(gtf[,7]),FUN=toString)\n",
    "rownames(groseq_count)=a2\n",
    "groseq_rpm=groseq_count\n",
    "for(j in a){\n",
    "    system(paste(\"echo\",j))\n",
    "temp=read.table(paste(pathGroseq,\"/\",j,sep=\"\"),sep=\"\\t\")\n",
    "tempGG=GRanges(seqnames=as.matrix(temp[,1]),ranges=IRanges(start = as.numeric(temp[,2]),end=as.numeric(temp[,3])),mcols=abs(as.numeric(temp[,4])))\n",
    "    for(i in seq(nrow(gtf))){\n",
    "        system(paste(\"echo\",i/nrow(gtf)))\n",
    "        gg=GRanges(seqnames=as.matrix(gtf[i,1]),ranges=IRanges(start = as.numeric(gtf[i,2]),end=as.numeric(gtf[i,3])))    \n",
    "        over=findOverlaps(gg,tempGG)@to    \n",
    "        name=as.vector(noExt((j)))\n",
    "        groseq_count[name,toString(gtf[i,7])]= sum(tempGG[over]$mcols)\n",
    "        groseq_rpm[name,toString(gtf[i,7])]= (sum(tempGG[over]$mcols)/sum(abs(temp[,4])))*1000000\n",
    "\n",
    "   }\n",
    "\n",
    "}\n",
    "write.table((t(groseq_count)),paste(results,\"/cutTagMet_AID_Heatmap_count.csv\",sep=\"\"),col.names=NA,sep=\",\")\n",
    "write.table((t(groseq_rpm)),paste(results,\"/cutTagMet_AID_Heatmap_RPM.csv\",sep=\"\"),col.names=NA,sep=\",\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e7f3a73-fc3f-422e-9147-f88824fd6609",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "269d3a4d-de46-4b26-a0f3-53a3ed982f3b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "832dcb42-7a59-46f2-be0b-f5562599ce1b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
