{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "750eb4a1-6e1a-4258-9e30-877259a6921e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Registered S3 method overwritten by 'ggtree':\n",
      "  method      from \n",
      "  identify.gg ggfun\n",
      "\n",
      "clusterProfiler v4.2.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141\n",
      "\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.2     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.4.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m   masks \u001b[34mclusterProfiler\u001b[39m::filter(), \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m      masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32msimplify()\u001b[39m masks \u001b[34mclusterProfiler\u001b[39m::simplify()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "DOSE v3.20.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "### GSEA enrichment analysis\n",
    "\n",
    "library(clusterProfiler)\n",
    "library(tidyverse)\n",
    "library(dplyr)\n",
    "library(msigdbr)\n",
    "library(enrichplot)\n",
    "library(DOSE)\n",
    "library(ggplot2)\n",
    "library(\"ggridges\")\n",
    "# GSEA_r <- snakemake@output[[\"GSEA_r\"]]\n",
    "# GSEA_ridgeplot <- snakemake@output[[\"GSEA_ridgeplot\"]]\n",
    "# type <- snakemake@wildcards[[\"msign\"]]\n",
    "# signature<-snakemake@wildcards[['cluster']]\n",
    "type='C8'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "5903740b-0e76-4d0f-9940-b63020823de0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sto facendo\"\n",
      "[1] \"sto facendo\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (14.24% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n",
      "Picking joint bandwidth of 0.814\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sto facendo\"\n"
     ]
    }
   ],
   "source": [
    "print(\"sto facendo\")\n",
    "waffle=\"/sharedFolder/Data/2_RNA_seq/TazeVSDMSO/DEfull.txt\"\n",
    "gene_res_df <- read.table(waffle, sep = \"\\t\",header = TRUE,row.names=1 )\n",
    "term2gene=rownames(gene_res_df)\n",
    "term2gene=t(rbind(sapply(term2gene,FUN=function(x){\n",
    "strsplit(x,\":\")[[1]]\n",
    "})))\n",
    "rownames(gene_res_df)=sapply(rownames(gene_res_df),FUN=function(x){\n",
    "strsplit(x,\":\")[[1]][2]\n",
    "})\n",
    "\n",
    "\n",
    "# Calcola l'indice di ordinamento\n",
    "sort_index <- order(-gene_res_df$stat)\n",
    "\n",
    "# Ordina la matrice in base all'indice di ordinamento\n",
    "gene_res_df <- gene_res_df[sort_index,]\n",
    "geneList=cbind(rownames(gene_res_df),gene_res_df$stat)\n",
    "geneList <- geneList[which(!is.na(geneList[,2])),]\n",
    "geneList2=as.numeric(geneList[,2])\n",
    "names(geneList2)=geneList[,1]\n",
    "sort_index <- order(geneList2, decreasing = TRUE)\n",
    "geneList2 <- geneList2[sort_index]\n",
    "print(\"sto facendo\")\n",
    "\n",
    "pdf(\"/sharedFolder/Data/2_RNA_seq/TazeVSDMSO/TAZENES_GO.pdf\")\n",
    "gse=gseGO(geneList2,ont=\"BP\",keyType=\"ENSEMBL\",OrgDb=\"org.Mm.eg.db\",eps=1e-300)\n",
    "ridgeplot(gse, showCategory = 20) +\n",
    "  theme(axis.text.y = element_text(size = 8)) +\n",
    "  labs(title = \"NES score DE Taze vs DMSO\")\n",
    "dev.off()\n",
    "result <- gse@result\n",
    "result$p.adjust=-log10(result$p.adjust)\n",
    "result <- result[sort(result$p.adjust,index.return=TRUE,decreasing=TRUE)$ix, ]\n",
    "result <- result[1:20, ]\n",
    "result$Description <- factor(result$Description, levels = result$Description)\n",
    "\n",
    "p <- ggplot(result, aes(x = p.adjust, y = reorder(Description, p.adjust))) +\n",
    "  geom_bar(stat = \"identity\", fill = \"red\") +\n",
    "  ylab(\"GO enrichment\") +\n",
    "  xlab(\"-log10(p.adjust)\") +\n",
    "  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))\n",
    "pdf(\"/sharedFolder/Data/2_RNA_seq/TazeVSDMSO/TAZEpadj_GeneOntology.pdf\")\n",
    "print(p)\n",
    "dev.off()\n",
    "print(\"sto facendo\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "337cd846-18e3-46b8-9d50-3708348c7348",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sto facendo\"\n",
      "[1] \"sto facendo\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (14.84% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n",
      "Picking joint bandwidth of 0.699\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"sto facendo\"\n"
     ]
    }
   ],
   "source": [
    "print(\"sto facendo\")\n",
    "waffle=\"/sharedFolder/Data/2_RNA_seq/ValeVSDMSO/DEfull.txt\"\n",
    "gene_res_df <- read.table(waffle, sep = \"\\t\",header = TRUE,row.names=1 )\n",
    "term2gene=rownames(gene_res_df)\n",
    "term2gene=t(rbind(sapply(term2gene,FUN=function(x){\n",
    "strsplit(x,\":\")[[1]]\n",
    "})))\n",
    "rownames(gene_res_df)=sapply(rownames(gene_res_df),FUN=function(x){\n",
    "strsplit(x,\":\")[[1]][2]\n",
    "})\n",
    "\n",
    "\n",
    "# Calcola l'indice di ordinamento\n",
    "sort_index <- order(-gene_res_df$stat)\n",
    "\n",
    "# Ordina la matrice in base all'indice di ordinamento\n",
    "gene_res_df <- gene_res_df[sort_index,]\n",
    "geneList=cbind(rownames(gene_res_df),gene_res_df$stat)\n",
    "geneList <- geneList[which(!is.na(geneList[,2])),]\n",
    "geneList2=as.numeric(geneList[,2])\n",
    "names(geneList2)=geneList[,1]\n",
    "sort_index <- order(geneList2, decreasing = TRUE)\n",
    "geneList2 <- geneList2[sort_index]\n",
    "print(\"sto facendo\")\n",
    "\n",
    "pdf(\"/sharedFolder/Data/2_RNA_seq/ValeVSDMSO/VALENES_GO.pdf\")\n",
    "gse=gseGO(geneList2,ont=\"BP\",keyType=\"ENSEMBL\",OrgDb=\"org.Mm.eg.db\",eps=1e-300)\n",
    "ridgeplot(gse, showCategory = 20) +\n",
    "  theme(axis.text.y = element_text(size = 8)) +\n",
    "  labs(title = \"NES score DE Vale vs DMSO\")\n",
    "dev.off()\n",
    "result <- gse@result\n",
    "result$p.adjust=-log10(result$p.adjust)\n",
    "result <- result[sort(result$p.adjust,index.return=TRUE,decreasing=TRUE)$ix, ]\n",
    "result <- result[1:20, ]\n",
    "result$Description <- factor(result$Description, levels = result$Description)\n",
    "\n",
    "p <- ggplot(result, aes(x = p.adjust, y = reorder(Description, p.adjust))) +\n",
    "  geom_bar(stat = \"identity\", fill = \"red\") +\n",
    "  ylab(\"GO enrichment\") +\n",
    "  xlab(\"-log10(p.adjust)\") +\n",
    "  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))\n",
    "pdf(\"/sharedFolder/Data/2_RNA_seq/ValeVSDMSO/Valepadj_GeneOntology.pdf\")\n",
    "print(p)\n",
    "dev.off()\n",
    "print(\"sto facendo\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "974d101c-6164-47cc-836f-fe4ec876c964",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.1.1 (2021-08-10)\n",
       "Platform: x86_64-pc-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 20.04.3 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS:   /usr/local/lib/R/lib/libRblas.so\n",
       "LAPACK: /usr/local/lib/R/lib/libRlapack.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=C.UTF-8    LC_NUMERIC=C        LC_TIME=C          \n",
       " [4] LC_COLLATE=C        LC_MONETARY=C       LC_MESSAGES=C      \n",
       " [7] LC_PAPER=C          LC_NAME=C           LC_ADDRESS=C       \n",
       "[10] LC_TELEPHONE=C      LC_MEASUREMENT=C    LC_IDENTIFICATION=C\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       " [1] ggridges_0.5.4        DOSE_3.20.1           enrichplot_1.14.2    \n",
       " [4] msigdbr_7.5.1         lubridate_1.9.2       forcats_1.0.0        \n",
       " [7] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.1          \n",
       "[10] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         \n",
       "[13] ggplot2_3.4.2         tidyverse_2.0.0       clusterProfiler_4.2.2\n",
       "[16] repr_1.1.6           \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       "  [1] fgsea_1.20.0           colorspace_2.1-0       ggtree_3.2.1          \n",
       "  [4] IRdisplay_1.1          qvalue_2.26.0          XVector_0.34.0        \n",
       "  [7] base64enc_0.1-3        aplot_0.1.10           farver_2.1.1          \n",
       " [10] graphlayouts_1.0.0     ggrepel_0.9.3          bit64_4.0.5           \n",
       " [13] AnnotationDbi_1.56.2   fansi_1.0.4            scatterpie_0.2.1      \n",
       " [16] splines_4.1.1          cachem_1.0.8           GOSemSim_2.20.0       \n",
       " [19] polyclip_1.10-4        IRkernel_1.3.2         jsonlite_1.8.7        \n",
       " [22] GO.db_3.14.0           png_0.1-8              ggforce_0.4.1         \n",
       " [25] compiler_4.1.1         httr_1.4.6             Matrix_1.6-0          \n",
       " [28] fastmap_1.1.1          lazyeval_0.2.2         cli_3.6.1             \n",
       " [31] tweenr_2.0.2           htmltools_0.5.5        tools_4.1.1           \n",
       " [34] igraph_1.5.0           gtable_0.3.3           glue_1.6.2            \n",
       " [37] GenomeInfoDbData_1.2.7 reshape2_1.4.4         DO.db_2.9             \n",
       " [40] fastmatch_1.1-3        Rcpp_1.0.11            Biobase_2.54.0        \n",
       " [43] vctrs_0.6.3            Biostrings_2.62.0      babelgene_22.9        \n",
       " [46] ape_5.7-1              nlme_3.1-162           ggraph_2.1.0          \n",
       " [49] timechange_0.2.0       lifecycle_1.0.3        zlibbioc_1.40.0       \n",
       " [52] MASS_7.3-60            scales_1.2.1           tidygraph_1.2.3       \n",
       " [55] hms_1.1.3              parallel_4.1.1         RColorBrewer_1.1-3    \n",
       " [58] memoise_2.0.1          gridExtra_2.3          downloader_0.4        \n",
       " [61] ggfun_0.1.1            yulab.utils_0.0.6      stringi_1.7.12        \n",
       " [64] RSQLite_2.3.1          S4Vectors_0.32.4       tidytree_0.4.2        \n",
       " [67] BiocGenerics_0.40.0    BiocParallel_1.28.3    GenomeInfoDb_1.30.1   \n",
       " [70] rlang_1.1.1            pkgconfig_2.0.3        bitops_1.0-7          \n",
       " [73] evaluate_0.21          lattice_0.21-8         treeio_1.18.1         \n",
       " [76] patchwork_1.1.2        shadowtext_0.1.2       bit_4.0.5             \n",
       " [79] tidyselect_1.2.0       plyr_1.8.8             magrittr_2.0.3        \n",
       " [82] R6_2.5.1               IRanges_2.28.0         generics_0.1.3        \n",
       " [85] pbdZMQ_0.3-9           DBI_1.1.3              pillar_1.9.0          \n",
       " [88] withr_2.5.0            KEGGREST_1.34.0        RCurl_1.98-1.12       \n",
       " [91] crayon_1.5.2           uuid_1.1-0             utf8_1.2.3            \n",
       " [94] tzdb_0.4.0             viridis_0.6.3          grid_4.1.1            \n",
       " [97] data.table_1.14.8      blob_1.2.4             digest_0.6.33         \n",
       "[100] gridGraphics_0.5-1     stats4_4.1.1           munsell_0.5.0         \n",
       "[103] viridisLite_0.4.2      ggplotify_0.1.1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "ceb3424e-dc07-449c-a1bc-984cd3c291f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#library(R.utils)\n",
    "#R.utils::setOption(\"clusterProfiler.download.method\",\"auto\")\n",
    "#getOption(\"clusterProfiler.download.method\")\n",
    "#gse <- gseKEGG(geneList2, organism=\"mmu\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8550c58b-eb20-41b6-9a28-ab23c46cbdc8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe3c7531-a83c-4465-a582-c584baf623f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a20a9f0f-adc3-4303-9a8e-32c642270713",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "617a1ec2-fea1-440b-bfa0-d95211bc50ea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef2975e7-0e98-411b-8d14-63aee89fd2be",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6a31e56-0b30-48d0-9e43-e40747ca8f1d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98cd283c-25b0-42c2-a3bf-3fda398f10e6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
